d2/src/main/java/com/linkedin/d2/balancer/util/downstreams/FSBasedDownstreamServicesFetcher.java

```diff
@@ -30,18 +33,34 @@
 public class FSBasedDownstreamServicesFetcher implements DownstreamServicesFetcher
 {
   private final String _d2FsPath;
+  private final String _indisFsPath;
   private final String _d2ServicePath;
 
   public FSBasedDownstreamServicesFetcher(String d2FsPath, String d2ServicePath)
+  {
+    this(d2FsPath, null, d2ServicePath);
+  }
+
+  public FSBasedDownstreamServicesFetcher(String d2FsPath, String indisFsPath, String d2ServicePath)
   {
     _d2FsPath = d2FsPath;
+    _indisFsPath = indisFsPath;
     _d2ServicePath = d2ServicePath;
   }
 
   @Override
   public void getServiceNames(SuccessCallback<List<String>> callback)
   {
+
     FileSystemDirectory fsDirectory = new FileSystemDirectory(_d2FsPath, _d2ServicePath);
-    callback.onSuccess(fsDirectory.getServiceNames());
+    Set<String> serviceNames = new HashSet<>(fsDirectory.getServiceNames());
+
+    if (_indisFsPath != null)
+    {
+      FileSystemDirectory indisFsDirectory = new FileSystemDirectory(_indisFsPath, _d2ServicePath);
+      serviceNames.addAll(indisFsDirectory.getServiceNames());
```

Comment:
Maybe I missed something, would mixing the services in indis dir and the ones in d2 dir actually cause the d2 simple load balancer tries to warm up some service that only exists in the indis file? and vice versa?

Or even that happens, it doesn't matter (just fails and wait for the ZK/observer to connect)? If so, could you add a comment here to explain it?

Reply:
Sounds good, will use a separate warm up load balancer for the INDIS path